Build failure with dbus crate

I'm seeing the following error when trying to build this on Fedora 37

````
   Compiling parsec-client v0.14.0
warning: unused import: `dbus::arg`
 --> /builddir/build/BUILD/dbus-parsec-0.3.1/target/release/build/dbus-parsec-361de4b5cf7b1530/out/dbus_parsec_control_client.rs:3:5
  |
3 | use dbus::arg;
  |     ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
warning: unused import: `parsec_client::core::secrecy::Secret`
  --> src/daemon.rs:23:5
   |
23 | use parsec_client::core::secrecy::Secret;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default
warning: unused import: `dbus::arg`
 --> /builddir/build/BUILD/dbus-parsec-0.3.1/target/release/build/dbus-parsec-361de4b5cf7b1530/out/dbus_parsec_control_server.rs:3:5
  |
3 | use dbus::arg;
  |     ^^^^^^^^^
warning: unused import: `dbus::arg`
 --> /builddir/build/BUILD/dbus-parsec-0.3.1/target/release/build/dbus-parsec-361de4b5cf7b1530/out/dbus_parsec_control_client.rs:3:5
  |
3 | use dbus::arg;
  |     ^^^^^^^^^
error[E0308]: mismatched types
  --> src/agent/control.rs:38:56
   |
38 |         match self.parsec_client.psa_export_public_key(key_name.clone()) {
   |                                                        ^^^^^^^^^^^^^^^^
   |                                                        |
   |                                                        expected `&str`, found struct `std::string::String`
   |                                                        help: consider borrowing here: `&key_name`
error[E0308]: mismatched types
  --> src/agent/control.rs:48:48
   |
48 |                         .psa_export_public_key(key_name.clone())
   |                                                ^^^^^^^^^^^^^^^^
   |                                                |
   |                                                expected `&str`, found struct `std::string::String`
   |                                                help: consider borrowing here: `&key_name`
error[E0308]: mismatched types
   --> src/agent/mod.rs:162:31
    |
162 |             .psa_generate_key(key_name.to_string(), key_attrs)?;
    |                               ^^^^^^^^^^^^^^^^^^^^
    |                               |
    |                               expected `&str`, found struct `std::string::String`
    |                               help: consider borrowing here: `&key_name.to_string()`
error[E0308]: mismatched types
   --> src/agent/mod.rs:211:41
    |
211 |                 .psa_asymmetric_decrypt(key_name, asym_enc_algo, wrapkey, None)
    |                                         ^^^^^^^^
    |                                         |
    |                                         expected `&str`, found struct `std::string::String`
    |                                         help: consider borrowing here: `&key_name`
For more information about this error, try `rustc --explain E0308`.
warning: `dbus-parsec` (bin "dbus-parsec") generated 3 warnings
error: could not compile `dbus-parsec` due to 4 previous errors; 3 warnings emitted
warning: build failed, waiting for other jobs to finish...
warning: `dbus-parsec` (bin "dbus-parsec-control") generated 1 warning
error: build failed
````

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ path = "src/utility.rs"
 [dependencies]
 dbus = "0.8.4"
 dbus-crossroads = "0.2.1"
-parsec-client = "0.11.0"
+parsec-client = "0.14.1"
 ring = { version = "0.16.15", features = ["std"] }
 anyhow = "1.0.32"
 rsa = "0.5.0"

--- a/src/agent/control.rs
+++ b/src/agent/control.rs
@@ -35,7 +35,7 @@ impl dbus_parsec_control_server::ComGithubPuiterwijkDBusPARSECControl for Agent 
             Some(key_id) => self.key_name(&key_id, secret_group),
         };
 
-        match self.parsec_client.psa_export_public_key(key_name.clone()) {
+        match self.parsec_client.psa_export_public_key(&key_name) {
             Ok(key) => Ok(key),
             Err(parsec_client::error::Error::Service(
                 parsec_client::core::interface::requests::ResponseStatus::PsaErrorDoesNotExist,
@@ -45,7 +45,7 @@ impl dbus_parsec_control_server::ComGithubPuiterwijkDBusPARSECControl for Agent 
                 match self.create_key(&key_name) {
                     Ok(_) => self
                         .parsec_client
-                        .psa_export_public_key(key_name.clone())
+                        .psa_export_public_key(&key_name)
                         .map_err(|err| {
                             MethodErr::failed(&format!(
                                 "Unexpected error getting key after creating: {}",

--- a/src/agent/mod.rs
+++ b/src/agent/mod.rs
@@ -137,29 +137,20 @@ impl Agent {
             hash_alg: Hash::Sha256,
         };
 
+        let mut usage_flags = UsageFlags::default();
+        usage_flags.set_encrypt().set_decrypt();
+
         let key_attrs = Attributes {
             lifetime: Lifetime::Persistent,
             key_type: Type::RsaKeyPair,
             bits: 2048,
             policy: Policy {
-                usage_flags: UsageFlags {
-                    export: false,
-                    copy: false,
-                    cache: false,
-                    encrypt: true,
-                    decrypt: true,
-                    sign_message: false,
-                    verify_message: false,
-                    sign_hash: false,
-                    verify_hash: false,
-                    derive: false,
-                },
+                usage_flags: usage_flags,
                 permitted_algorithms: asym_enc_algo.into(),
             },
         };
 
-        self.parsec_client
-            .psa_generate_key(key_name.to_string(), key_attrs)?;
+        self.parsec_client.psa_generate_key(key_name, key_attrs)?;
 
         Ok(())
     }
@@ -208,7 +199,7 @@ impl Agent {
         let wrapkey =
             match self
                 .parsec_client
-                .psa_asymmetric_decrypt(key_name, asym_enc_algo, wrapkey, None)
+                .psa_asymmetric_decrypt(&key_name, asym_enc_algo, wrapkey, None)
             {
                 Ok(key) => key,
                 Err(err) => {

--- a/src/agent/networkmanager.rs
+++ b/src/agent/networkmanager.rs
@@ -111,9 +111,16 @@ impl<'a> NMConnectionInfo<'a> for &NMConnection<'a> {
                 self.get("vpn")?.get("service-type")?.0.as_str()?,
             )),
             "wireguard" => Some(NetworkManagerConnectionType::Wireguard),
-            "802-11-wireless" => match self.get("802-11-wireless-security")?.get("key-mgmt")?.0.as_str()? {
+            "802-11-wireless" => match self
+                .get("802-11-wireless-security")?
+                .get("key-mgmt")?
+                .0
+                .as_str()?
+            {
                 "wpa-psk" => Some(NetworkManagerConnectionType::Wireless(NMWirelessType::PSK)),
-                _ => Some(NetworkManagerConnectionType::Wireless(NMWirelessType::Enterprise)),
+                _ => Some(NetworkManagerConnectionType::Wireless(
+                    NMWirelessType::Enterprise,
+                )),
             },
             _ => None,
         }

--- a/src/daemon.rs
+++ b/src/daemon.rs
@@ -20,7 +20,6 @@ use dbus::tree;
 use dbus::tree::{Factory, Interface, MTFn};
 
 use parsec_client::core::interface::requests::Opcode;
-use parsec_client::core::secrecy::Secret;
 use parsec_client::BasicClient;
 
 mod agent;

